services/algolia-sync/src/components/algolia/message-sync.js

```diff
@@ -37,6 +38,11 @@ const upsertToIndex = async (message) => {
 
     c.boost = boostResult(c);
 
+    // Set unpublished date to 100 years in the future if it's null
+    if (c.unpublishedDate === null) {
```

Comment:
Change `unpublishedDate` references to `unpublished`

Also, change your if statement to `if (!c.unpublished)`